pennylane/pauli/pauli_arithmetic.py

```diff
@@ -55,17 +55,30 @@
     Z: matZ,
 }
 
-sparse_matI = sparse.eye(2, format="csr")
-sparse_matX = sparse.csr_matrix([[0, 1], [1, 0]])
-sparse_matY = sparse.csr_matrix([[0, -1j], [1j, 0]])
-sparse_matZ = sparse.csr_matrix([[1, 0], [0, -1]])
 
-sparse_mat_map = {
-    I: sparse_matI,
-    X: sparse_matX,
-    Y: sparse_matY,
-    Z: sparse_matZ,
-}
+@lru_cache
+def _cached_sparse_data(op):
+    """Returns the sparse data and indices of a Pauli operator."""
+    if op == "I":
+        data = np.array([1.0, 1.0], dtype=np.complex128)
+        indices = np.array([0, 1], dtype=np.int64)
+    elif op == "X":
+        data = np.array([1.0, 1.0], dtype=np.complex128)
+        indices = np.array([1, 0], dtype=np.int64)
+    elif op == "Y":
+        data = np.array([-1.0j, 1.0j], dtype=np.complex128)
+        indices = np.array([1, 0], dtype=np.int64)
+    elif op == "Z":
+        data = np.array([1.0, -1.0], dtype=np.complex128)
+        indices = np.array([0, 1], dtype=np.int64)
+    return data, indices
+
+
+@lru_cache
```

Comment:
To avoid requiring too much memory
```suggestion
@lru_cache(maxsize=2)
```